editor/plugins/script_editor_plugin.cpp

```diff
@@ -4457,7 +4457,7 @@ ScriptEditor::ScriptEditor(WindowWrapper *p_wrapper) {
 	add_child(help_search_dialog);
 	help_search_dialog->connect("go_to_help", callable_mp(this, &ScriptEditor::_help_class_goto));
 
-	find_in_files_dialog = memnew(FindInFilesDialog);
+	find_in_files_dialog = FindInFilesDialog::get_singleton();
```

Comment:
Find in Files is still dependent on ScriptEditor to set it up first, so it shouldn't be a Singleton by itself.
We should either move Find in Files to its own plugin somehow, or `_on_find_in_files_requested` can just be made public, and the EditorNode can call it on the ScriptEditor singleton.

Reply:
To confirm my understanding your latter suggestion is saying that anywhere else that accesses the FindInFilesDialog (currently only EditorNode), should do it through the ScriptEditor singleton because `_on_find_in_files_requested` becomes public, and this would make it so FindInFilesDialog doesn't need to be a singleton anymore.

I'm not too clear on the suggestion of its own plugin, so the above suggestion seems clear to me. Happy to have some guidance/discuss further

Reply:
Actually, I meant that the EditorNode should call `ScriptEditor::get_singleton()->on_find_in_files_requested` and FindInFilesDialog is not made a Singleton.
The alternative of making FindInFiles a separate plugin would require separating it from the ScriptEditor as much as possible, and it would take more effort. But for this PR it's probably best to just make `_on_find_in_files_requested` public.

Reply:
thanks for the clarification, I misunderstood and changed my original post, will make an update now